### PR TITLE
initial project structure and implementation of HttpTest classes

### PR DESCRIPTION
**Pull request by eli-darkly**
This PR creates our standard .NET library repo structure, and populates this new package with an HTTP testing tool adapted from the one that we're currently using in the Xamarin SDK. Somewhat different implementations of these tools currently exist in our other .NET projects, and the plan is to migrate them all to this.

The `HttpTest` namespace provides a simple portable abstraction for an embedded HTTP test server with programmable behavior. This allows testing of HTTP client logic in a more realistic way than the basic "write a custom HttpMessageHandler that pretends to do a request" approach. The API is based on a simple request handler model, similar to what we've used in our Go and Node.js projects; tests that require more complex server behavior can add simple routing logic within a custom handler.

Unlike Go and Node, .NET doesn't have a built-in high-level implementation of a basic web server— at least, not on all the platforms we need to support; in .NET Framework there is an ASP.NET server API, but otherwise you need to use the lower-level `HttpListener` API. That's not very convenient, so .NET developers have created several higher-level testing frameworks such as [EmbedIO](https://github.com/unosquare/embedio) and [WireMock.Net](https://github.com/WireMock-Net/WireMock.Net), which we've used before; for various reasons (EmbedIO doesn't have a build for .NET Framework, WireMock.Net can't do streaming), none of those would work for all of our test scenarios. But they all used `HttpListener` either directly or indirectly, so it wasn't hard to pick out the basic approach that they all had in common and use it as the basis for this.

Migrating our projects to use this API should be a good thing in terms of consistent implementation, better test abstraction, and more complete cross-platform test coverage.

### CODE REVIEW

File: src/LaunchDarkly.TestHelpers/HttpTest/Handlers.cs [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    /// <param name="context">the request context</param>
+    /// <returns>the asynchronous task</returns>
+    public delegate Task Handler(IRequestContext context);
+
+    /// <summary>
+    /// Factory methods for standard <see cref="Handler"/> implementations.
+    /// </summary>
+    public static class Handlers
+    {
+        //#pragma warning disable CS1998 // allow async methods with no await
```
Comment (gwhelanLD) [inline src/LaunchDarkly.TestHelpers/HttpTest/Handlers.cs:24]: Probably unintended comment left here.

File: src/LaunchDarkly.TestHelpers/HttpTest/Handlers.cs [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+        /// <summary>
+        /// Creates a <see cref="Handler"/> that sleeps for the specified amount of time.
+        /// </summary>
+        /// <param name="delay">how long to delay</param>
+        /// <returns>a <see cref="Handler"/></returns>
+        public static Handler Delay(TimeSpan delay) =>
+            async ctx => await Task.Delay(delay, ctx.CancellationToken);
+
+        /// <summary>
+        /// Creates a <see cref="Handler"/> that sleeps indefinitely, holding the conneciton open,
```
Comment (gwhelanLD) [inline src/LaunchDarkly.TestHelpers/HttpTest/Handlers.cs:189]: Typo: conneciton -> connection

**Review by gwhelanLD — CHANGES_REQUESTED** [review]
The only significant note is the releaser configuration publish URL. Otherwise looks good.

File: .ldrelease/config.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+publications:
+  - url: https://nuget.org/packages/LaunchDarkly.EventSource
```
[the remark this answers is not in the document]
Comment (eli-darkly) [reply .ldrelease/config.yml:3]: Whoops, copypasta

**Review by gwhelanLD — APPROVED** [review]

**Pull request merged**